llarp/link/server.cpp

```diff
@@ -175,16 +179,22 @@ namespace llarp
   {
     std::vector< util::StatusObject > pending, established;
 
-    std::transform(m_Pending.begin(), m_Pending.end(),
```

Comment:
this is have been a data race[1], which when fixed would have become a potential deadlock.

[1] - if this code wasn't using `NullLock`